tests/flow/test_algebraic_expression_order.py

```diff
@@ -0,0 +1,74 @@
+import os
+import sys
+from RLTest import Env
+from redisgraph import Graph, Node, Edge
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+
+from base import FlowTestsBase
+
+graph = None
+
+GRAPH_ID = "AlgebraicExpressionOrder"
+
+class testAlgebraicExpressionOrder(FlowTestsBase):
+    def __init__(self):
+        self.env = Env()
+        global graph
+        redis_con = self.env.getConnection()
+        graph = Graph(GRAPH_ID, redis_con)
+        self.populate_graph()
+
+    def populate_graph(self):
+        # Construct a graph with the form:
+        # (v1:A)-[:E]->(v2:B), (v3:C)-[:E]->(v2)
+        v1 = Node(label="A", properties={"v": 1})
+        graph.add_node(v1)
+
+        v2 = Node(label="B", properties={"v": 2})
+        graph.add_node(v2)
+
+        v3 = Node(label="C", properties={"v": 3})
+        graph.add_node(v3)
+
+        edge = Edge(v1, "E", v2)
+        graph.add_edge(edge)
+
+        edge = Edge(v3, "E", v2)
+        graph.add_edge(edge)
+
+        graph.commit()
+
+    # Test differing patterns with the same destination node.
+    def test01_same_destination_permutations(self):
```

Comment:
Just for completeness - please add the scenarios:
1. both src and dest are labeled - dest is filtered (as in #1288)
1. both src and dest are labeled - src is filtered